### ISSUE TITLE
Community activity is not being updated

### ISSUE THREAD

**Issue by hvelarde** [plone/ploneorg.core#180]
![selection_073](https://cloud.githubusercontent.com/assets/341393/15782961/a190a83c-2982-11e6-8545-42ee46c78f2e.png)

**Comment by tkimnguyen** [comment]
@sneridagh do you know the status of these github data scripts?

**Comment by sneridagh** [comment]
@tkimnguyen Al I can say is that the stats retriever definitellly needs more love. I was debugging it last week and found some errors on the Twitter and SO parts. I'm half way on these ones.

No time to look into why the Github related ones are failing (or if they aren't, why they show always 0).

**Comment by hvelarde** [comment]
we can split this into 5 different issues; maybe I can help if you point me to the code.

**Comment by sneridagh** [comment]
Sure! Thanks!
The stats retriever consists in the retriever itself, that gets everything from this script:
https://github.com/plone/ploneorg.core/blob/master/ploneorg/core/contributions.py

And the related Plone views counterparts:
https://github.com/plone/ploneorg.core/blob/master/ploneorg/core/browser/contributor.py#L162
https://github.com/plone/ploneorg.core/blob/master/ploneorg/core/browser/contributor.py#L328
https://github.com/plone/ploneorg.core/blob/master/ploneorg/core/browser/contributor.py#L342

That loads the retrieved information in the right place in Plone.

**Comment by hvelarde** [comment]
@sneridagh can you explain the logic? I think the way is implemented is quite complicated and difficult to debug as it only works on the plone.org site itself.

the `contributions.py` script should run standalone.

**Comment by hvelarde** [comment]
well, it seems this is worst than I was expecting: can someone explain me how exactly are we going to measure the contributions?

* what is **Plone contributions**? the sum of all of the below? something else?
* **Plone commits** and **Collective commits** seem to be clear, but is not going to be easy to achieve (see [my blog post from 2014](http://hvelarde.blogspot.com/2014/01/how-to-get-statistics-about-your.html))
* what are you trying to measure in **StackOverflow**? [answers by tag](http://api.stackexchange.com/2.2/users/644075/top-answer-tags?site=stackoverflow)? [top "plone" answerers in last month](http://api.stackexchange.com/2.2/tags/plone/top-answerers/month?site=stackoverflow)?
* **Tweets about**? total? on a period? [the API](https://dev.twitter.com/rest/reference/get/search/tweets) doesn't give you an easy answer.

I think we put the cart (design) before the horse (implementation) in this case.

**Comment by davilima6** [comment]
We can adopt Github's policy for counting contributions: https://help.github.com/articles/why-are-my-contributions-not-showing-up-on-my-profile/ We can start with that and later evolve to integrate other APIs into the average/total.

**Comment by tkimnguyen** [comment]
OK, fair questions, but at the moment let's just get what we have working... we can refine and revisit what we're counting later.

**Comment by hvelarde** [comment]
@tkimnguyen I think I was not clear enough: we can't fix this without redefining what we want to get.

**Comment by tkimnguyen** [comment]
ping :)

**Comment by tkimnguyen** [comment]
Maybe we should hide this section until we have the stats working correctly?

**Comment by gforcada** [comment]
I see that all but blockers have some non-zero number, should this be closed then?

**Comment by tkimnguyen** [comment]
It's showing only 2 downloads of Plone in the last month :( 

![badactivity](https://cloud.githubusercontent.com/assets/815418/24522299/c2e927ca-1554-11e7-9081-c67775f49f65.png)

**Comment by sneridagh** [comment]
Indeed, we have to find time to give it some love…

**Comment by hvelarde** [comment]
no, we need to remove the feature and never include half-backed features in the future anymore. someone has to say it clearly ;-)

**Comment by tkimnguyen** [comment]
Help us fix it, @hvelarde

**Comment by hvelarde** [comment]
it's not that easy as it requires some time, but it could be a nice experiment for your students ;-)

**Comment by gforcada** [comment]
Sorry @hvelarde but saying it *clearly* and giving stop energy is a complete different story. I did work quite a lot on that to now get a "half backed features". It's really late here (1AM) so I'm not in my best mood either, but reading these kind of comments with finger pointing and  giving instructions is not the way forward either.

Some conditionals (*it could be removed if*) would be far more nicer and respectful than slamming a smile at the end of one sentences. See :-)

We are all overworked and are doing a lot on our free time, so let's appreciate that and try to find budgets to get paid time to work on thing if we want to be as professional looking as we all want.

**Comment by hvelarde** [comment]
@gforcada I disagree: I put energy on this last year and this has been broken since even before; the best way to solve problems is stop creating them in the first place.

this was a nice idea, yes; but if it was not completely implemented it should had never go into production; we have more important things to fix and I'm not going to spend time on this, not now, and not later.

and, IMO, you don't need to be awake up to 1AM fixing things like this, seriously; get some sleep and we talk tomorrow when you (and I) are on best mood ;-)

**Comment by hvelarde** [comment]
and, BTW, the statistics I was reporting are the ones on the people's profile and they continue to be on zero... all of them... and they haven't been defined neither:

![selection_022](https://cloud.githubusercontent.com/assets/341393/24530127/443d5b62-1586-11e7-83cc-5a9caf92e329.png)

**Comment by polyester** [comment]
It might be advantageous to look at http://grimoirelab.github.io, the components can be used independently and are updated much more often than we can/should do.

**Comment by hvelarde** [comment]
@polyester that sounds interesting but, as mentioned previously, we have to define what's the activity we're going to track; you can't implement a feature if the feature is not defined and you can't add a feature in production, if the feature is not working.

**Comment by sneridagh** [comment]
Hi,
So we have this on the home page:
[image: Screen Shot 2017-04-05 at 19.58.46.png]
agreed that we can narrow these ones a little bit. Like, let's say last
year/month/week... It's nonsense to have it since forever (and it's not
clear that Github is able to provide those).

and this in the user profile:
[image: Screen Shot 2017-04-05 at 19.59.25.png]
I still think that they are doable, however, they pose some challenges, the
most important one, the query rate limitations in each of the third party
services (that still apply on the Grimoirelab tools) which I think they are
pretty promissing (and Python based :P The second challenge is to track all
the repos that we have both in Plone organization and in Collective.

For these ones, I would try to get the most important ones, then hide the
ones that we don't have it.

What do you think? Hector, any suggestion on what items do you leave and
which one would you remove?

Cheers,
V.

**Comment by hvelarde** [comment]
@sneridagh do you mind to upload the images again? I can't see them.

**Comment by sneridagh** [comment]
<img width="1201" alt="screen shot 2017-04-05 at 19 58 46" src="https://cloud.githubusercontent.com/assets/486927/24722403/e5c49632-1a43-11e7-98b9-c09e9e7aa323.png">
<img width="974" alt="screen shot 2017-04-05 at 19 59 25" src="https://cloud.githubusercontent.com/assets/486927/24722405/e7d87272-1a43-11e7-84bd-e38df5e6bd65.png">

**Comment by sneridagh** [comment]
I've been looking into the tools that @polyester suggested. They are indeed very nice! I think we can do something with them... However, to take full advantage of them we need setup an ELK backend. I will take a look at the whole thing.

@tkimnguyen , do you think that we could have an ELK stack in the new (Wildcard donated) machine? We can try to set it up in the plone.org machine but maybe it's too much for it.

**Comment by hvelarde** [comment]
this issue was opened to address the activity of users of the Plone site; I suggest the following:

* list only Plone contributions and tweets about Plone
* Plone contributions should reflect contributions in both the plone and collective organizations on GitHub in the last year
* tweets about Plone should show the number of tweets containing the word "Plone" in the last year

I don't know if that is achievable, but it's a start.

**Comment by tkimnguyen** [comment]
@sneridagh did a lot of digging and ultimately we realized that the download numbers are the only ones that are not being updated. So we totaled up the downloads from various sources and updated the number on plone.org. Will update that download number manually periodically.

![screen shot 2017-05-01 at may 1 6 23 01 am](https://cloud.githubusercontent.com/assets/815418/25578671/a6f9f652-2e36-11e7-8f54-071dd02e2504.jpg)
